docs/RFCS/explain_debug.md

```diff
@@ -0,0 +1,119 @@
+- Feature Name: explain_debug
+- Status: in-progress
+- Start Date: 2015-08-10
+- RFC PR: #2047
+- Cockroach Issue:
+
+# Summary
+
+Add support for an `EXPLAIN DEBUG` statement. The `EXPLAIN DEBUG` statement
+will cause simple (single table or index) select statements to display the
+underlying key-value pairs that were accessed:
+
+* the first column will contain the index of the row the key maps to.
+* the second column will contain a pretty-printed version of the key in the
+  underlying key-value store.
+* the third column will contain a pretty-printed version of the value.
+* the fourth column will contain a boolean indicating whether the row would
+  have been included in the normal output or not.
+
+As an example, consider the following table:
+
+```sql
+CREATE TABLE foo (
+  a INT PRIMARY KEY,
+  b TEXT,
+  c BOOL,
+  CONSTRAINT bar UNIQUE (b)
+)
+```
+
+We'll populate the table with a couple of rows:
+
+```sql
+INSERT INTO foo VALUES (1, 'one', true), (2, 'two', false), (3, 'three', true)
+```
+
+`EXPLAIN DEBUG` can be used to show the key-value pairs accessed during the
```

Comment:
This output could be quite large; we should be able to limit its size (maybe just a hard cap instead of making room in the query to specify something).